### PR TITLE
managed vaults

### PR DESCRIPTION
**Pull request by vanillaHill**
WIP https://github.com/FortressFinance/issues/issues/77

### CODE REVIEW

File: src/components/Layout.tsx [hunk 1]
```diff
@@ -101,6 +105,24 @@ const Layout: FC<PropsWithChildren> = ({ children }) => {
...
+                              Managed Vaults
+                            </span>
+                            <span className="ml-1 grow-0 whitespace-nowrap rounded bg-pink-200/20 px-1 py-0.5 text-[9px] uppercase leading-tight text-pink-100/80">
+                              Coming soon
+                            </span>
+                          </span>
+                        ) : (
+                          <NavigationMenu.Link asChild>
+                            <DropdownMenuItemLink
+                              {...resolvedRoute("/app/yield/managed_vaults")}
```
Comment (aquaticone) [inline src/components/Layout.tsx:120]: Routes should use always hyphens instead of underscores.

File: src/pages/app/yield/managed_vaults.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    <Layout>
+      <Seo
+        templateTitle="Managed Vaults"
+        description="The Managed Vaults architecture provides permissionless primitives for flexible investment strategy management on-chain."
+      />
+
+      <main>
+        <Tabs.Root
+          value={
+            category && typeof category === "string" ? category : "featured"
```
Comment (aquaticone) [inline src/pages/app/yield/managed_vaults.tsx:29]: `typeof` check handles undefined/null. `category &&` is redundant.

**Review by aquaticone — CHANGES_REQUESTED** [review]

File: src/components/Modal/VaultStrategyModal/managedVaults/ManagedVaultsActivityTable.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        <span className="max-md:hidden">
+          {shortenAddress(activity.explorer)}
+        </span>
+        <span className="lg:hidden">
+          {shortenAddress(activity.explorer, 2)}
+        </span>
+      </div>
+    )
+  })
+  return <div className="max-h-[350px] w-full overflow-y-scroll">{results}</div>
```
Comment (aquaticone) [inline src/components/Modal/VaultStrategyModal/managedVaults/ManagedVaultsActivityTable.tsx:61]: Just map here no need for the array. `activities.map((activitiy) => <div></div>)`

File: src/lib/helpers/shortenAddress.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { getAddress } from "@ethersproject/address"
+
+// returns the checksummed address if the address is valid, otherwise returns false
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function isAddress(value: any): string | false {
```
Comment (aquaticone) [inline src/lib/helpers/shortenAddress.ts:5]: Just use `ethers.utils.isAddress`?

**Review by aquaticone — CHANGES_REQUESTED** [review]
I think this looks OK. A lot here so a little tough to take it all in but in general 👍

File: src/lib/helpers/shortenAddress.ts (hunk 4 above)
Comment (vanillaHill) [reply src/lib/helpers/shortenAddress.ts:5]: isAddress behaves differently of projectethers isAddress lib.

File: src/components/Modal/VaultStrategyModal/managedVaults/ManagedVaultsActivityTable.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (vanillaHill) [reply src/components/Modal/VaultStrategyModal/managedVaults/ManagedVaultsActivityTable.tsx:61]: I'll change it.

**Review by aquaticone — APPROVED** [review]

**Pull request merged**